### PR TITLE
cli: modernize docstrings

### PR DESCRIPTION
**Pull request by Exirel**
### Description

Part of #1565 for `sopel.cli`.

### Checklist

- [x] I have read [CONTRIBUTING.md](https://github.com/sopel-irc/sopel/blob/master/CONTRIBUTING.md)
- [x] I can and do license this contribution under the EFLv2
- [x] No issues are reported by `make qa` (runs `make quality` and `make test`)
- [x] I have tested the functionality of the things this change touches

### CODE REVIEW

File: sopel/cli/config.py [hunk 1]
```diff
@@ -66,6 +70,7 @@ def handle_list(options):
 
     :param options: parsed arguments
     :type options: ``argparse.Namespace``
+    :return: Return 0 if everything went fine.
```
Comment (dgw) [inline sopel/cli/config.py:73]:
```suggestion
    :return: 0 if everything went fine
```

This is our style elsewhere, in existing Sphinx docs.

Also, it's weird that no other return value (exit code) is possible.

File: sopel/cli/config.py [hunk 2]
```diff
@@ -103,6 +108,8 @@ def handle_init(options):
...
+             Return 1 if the file is invalid or if it already exists.
```
Comment (dgw) [inline sopel/cli/config.py:112]:
```suggestion
    :return: 0 if everything went fine;
             1 if the file is invalid or if it already exists
```

Style-matching again. Kept your wrapping style though. 😁

File: sopel/cli/config.py [hunk 3]
```diff
@@ -129,7 +136,14 @@ def handle_init(options):
...
 
 def handle_get(options):
-    """Read the settings to display the value of <section> <key>"""
+    """Read the settings to display the value of <section> <key>
+
+    :param options: parsed arguments
+    :type options: ``argparse.Namespace``
+    :return: Return 0 if everything went fine.
+             Return 1 if the section and/or option does not exist.
+             Return 2 if the settings can't be loaded
```
Comment (dgw) [inline sopel/cli/config.py:145]:
```suggestion
    :return: 0 if everything went fine;
             1 if the section and/or option does not exist;
             2 if the settings can't be loaded
```

This was already inconsistent with its siblings above—the last item had no trailing `.` 😹

File: sopel/cli/plugins.py [hunk 4]
```diff
@@ -149,7 +153,12 @@ def build_parser():
 
 
 def handle_list(options):
-    """List Sopel plugins"""
+    """List Sopel plugins
+
+    :param options: parsed arguments
+    :type options: ``argparse.Namespace``
+    :return: Return 0 if everything went fine.
```
Comment (dgw) [inline sopel/cli/plugins.py:160]:
```suggestion
    :return: 0 if everything went fine
```

Here again, weird that 0 is the only possible return value (exit code).

File: sopel/cli/plugins.py [hunk 5]
```diff
@@ -226,7 +235,13 @@ def handle_list(options):
 
 
 def handle_show(options):
-    """Show plugin details"""
+    """Show plugin details
+
+    :param options: parsed arguments
+    :type options: ``argparse.Namespace``
+    :return: Return 0 if everything went fine.
+             Return 1 otherwise. (plugin doesn't exist or can't be loaded)
```
Comment (dgw) [inline sopel/cli/plugins.py:243]:
```suggestion
    :return: 0 if everything went fine;
             1 if the plugin doesn't exist or can't be loaded
```

File: sopel/cli/plugins.py [hunk 6]
```diff
@@ -276,7 +291,13 @@ def handle_show(options):
 
 
 def handle_configure(options):
-    """Configure a Sopel plugin with a config wizard"""
+    """Configure a Sopel plugin with a config wizard
+
+    :param options: parsed arguments
+    :type options: ``argparse.Namespace``
+    :return: Return 0 if everything went fine.
+             Return 1 otherwise. (plugin doesn't exist or can't be loaded)
```
Comment (dgw) [inline sopel/cli/plugins.py:299]:
```suggestion
    :return: 0 if everything went fine;
             1 if the plugin doesn't exist or can't be loaded
```

File: sopel/cli/plugins.py [hunk 7]
```diff
@@ -349,7 +374,14 @@ def display_unknown_plugins(unknown_plugins):
...
 
 def handle_disable(options):
-    """Disable Sopel plugins"""
+    """Disable Sopel plugins
+
+    :param options: parsed arguments
+    :type options: ``argparse.Namespace``
+    :return: Return 0 if everything went fine.
+             Return 1 for coretasks (cannot be disabled)
+             or if the plugin doesn't exist.
```
Comment (dgw) [inline sopel/cli/plugins.py:383]:
```suggestion
    :return: 0 if everything went fine;
             1 if the plugin doesn't exist,
             or if attempting to disable coretasks (required)
```

I haven't test-rendered this to see what it looks like though.

File: sopel/cli/plugins.py [hunk 8]
```diff
@@ -450,7 +482,13 @@ def _handle_enable_plugin(settings, usable_plugins, plugin_name, allow_only):
 
 
 def handle_enable(options):
-    """Enable a Sopel plugin"""
+    """Enable a Sopel plugin
+
+    :param options: parsed arguments
+    :type options: ``argparse.Namespace``
+    :return: Return 0 if everything went fine.
+             Return 1 if the plugin doesn't exist.
```
Comment (dgw) [inline sopel/cli/plugins.py:490]:
```suggestion
    :return: 0 if everything went fine;
             1 if the plugin doesn't exist
```

File: sopel/cli/run.py [hunk 9]
```diff
@@ -49,6 +49,13 @@
 
 
 def run(settings, pid_file, daemon=False):
+    """Run the bot with these ``settings``
+
+    :param settings: bot's settings to run with
+    :type settings: :class:`sopel.config.Config`
+    :param str pid_file: path to the bot's PID file
+    :param bool daemon: tell if the bot should be ran as a daemon
```
Comment (dgw) [inline sopel/cli/run.py:57]:
```suggestion
    :param bool daemon: tell if the bot should be run as a daemon
```

File: sopel/cli/run.py [hunk 10]
```diff
@@ -49,6 +49,13 @@
 
 
 def run(settings, pid_file, daemon=False):
+    """Run the bot with these ``settings``
+
+    :param settings: bot's settings to run with
```
Comment (dgw) [inline sopel/cli/run.py:54]:
```suggestion
    :param settings: settings with which to run the bot
```

`/me English snob`

File: sopel/cli/run.py [hunk 11]
```diff
@@ -385,7 +401,11 @@ def get_running_pid(filename):
 
 
 def command_start(opts):
-    """Start a Sopel instance"""
+    """Start a Sopel instance
+
+    :param opts: parsed arguments
+    :type opts: ``argparse.Namespace``
```
Comment (dgw) [inline sopel/cli/run.py:407]: This goes for all the occurrences of this class: Is there no easy cross-reference for use with the Sphinx `:class:` role?

File: sopel/cli/run.py (hunk 10 above)
Comment (Exirel) [reply sopel/cli/run.py:54]: `/me French appreciation for snobiness`

File: sopel/cli/run.py (hunk 11 above)
Comment (Exirel) [reply sopel/cli/run.py:407]: I may have forgot that for some reason. (probably a lack of sleep)

**Comment by Exirel** [comment]
From the `CONTRIBUTING.md` file:

> A one-line description, as an imperative sentence (ending in a period)

It's on me.

File: sopel/cli/utils.py [hunk 12]
```diff
@@ -39,7 +39,7 @@ def _colored(text, color, reset=True):
 
 
 def green(text, reset=True):
-    """Add ANSI escape sequences to make the text green in term
+    """Add ANSI escape sequences to make the text green in term.
```
Comment (dgw) [inline sopel/cli/utils.py:42]: I'm going to add a batch of suggestions all related to this so you can apply (or ignore) them all easily.

Seems like something I should have flagged when these went in in the first place, but oh well.

```suggestion
    """Add ANSI escape sequences to make the text green in terminal.
```

File: sopel/cli/utils.py [hunk 13]
```diff
@@ -50,7 +50,7 @@ def green(text, reset=True):
 
 
 def yellow(text, reset=True):
-    """Add ANSI escape sequences to make the text yellow in term
+    """Add ANSI escape sequences to make the text yellow in term.
```
Comment (dgw) [inline sopel/cli/utils.py:53]:
```suggestion
    """Add ANSI escape sequences to make the text yellow in terminal.
```

File: sopel/cli/utils.py [hunk 14]
```diff
@@ -61,7 +61,7 @@ def yellow(text, reset=True):
 
 
 def red(text, reset=True):
-    """Add ANSI escape sequences to make the text red in term
+    """Add ANSI escape sequences to make the text red in term.
```
Comment (dgw) [inline sopel/cli/utils.py:64]:
```suggestion
    """Add ANSI escape sequences to make the text red in terminal.
```

**Comment by Exirel** [comment]
Rebased & squashed, as it's ready to be merged, unless @dgw finds new mistakes.

File: sopel/cli/config.py [hunk 15]
```diff
@@ -129,7 +136,14 @@ def handle_init(options):
 
 
 def handle_get(options):
-    """Read the settings to display the value of <section> <key>"""
+    """Read the settings to display the value of <section> <key>.
+
+    :param options: parsed arguments
+    :type options: :class:`argparse.Namespace`
+    :return: 0 if everything went fine;
+             1 if the section and/or option does not exist;
```
Comment (dgw) [inline sopel/cli/config.py:144]:
```suggestion
             1 if the section and/or key does not exist;
```

File: sopel/cli/config.py [hunk 16]
```diff
@@ -129,7 +136,14 @@ def handle_init(options):
 
 
 def handle_get(options):
-    """Read the settings to display the value of <section> <key>"""
+    """Read the settings to display the value of <section> <key>.
```
Comment (dgw) [inline sopel/cli/config.py:139]:
```suggestion
    """Read the settings to display the value of ``<section> <key>``.
```

Or should it be:

```suggestion
    """Read the settings to display the value of ``<section>`` ``<key>``.
```

It does seem weird not to monospace these.

**Comment by Exirel** [comment]
:shipit: :shipit: :shipit: :shipit:

**Pull request merged**